include/acquire.zarr.h

```diff
@@ -0,0 +1,274 @@
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_reserve_dimensions(
+      ZarrStreamSettings* settings,
+      size_t count);
+
+    /**
+     * @brief Set properties for an acquisition dimension.
+     * @detail The order of the dimensions in the Zarr stream is the order in
+     * which they are set. The first dimension set is the slowest varying
+     * dimension, and the last dimension set is the fastest varying dimension.
+     * For example, if the dimensions are set in the order z, y, x, the fastest
+     * varying dimension is x, the next fastest varying dimension is y, and the
+     * slowest varying dimension is z.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] index The index of the dimension to set. Must be less than the
+     * number of dimensions reserved with ZarrStreamSettings_reserve_dimensions.
+     * @param[in] dimension The dimension's settings.
+     */
+    ZarrStatus ZarrStreamSettings_set_dimension(
+      ZarrStreamSettings* settings,
+      size_t index,
+      const ZarrDimensionProperties* dimension);
+
+    /**
+     * @brief Set the multiscale flag for the Zarr stream.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param[in] multiscale A flag indicating whether to stream to multiple
+     * levels of detail.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_multiscale(ZarrStreamSettings* settings,
+                                                 uint8_t multiscale);
+
+    /**
+     * @brief Set JSON-formatted custom metadata for the Zarr stream.
+     * @details This metadata will be written to acquire-zarr.json in the
+     * metadata directory of the Zarr store. This parameter is optional.
+     * @param settings[in, out] settings The Zarr stream settings struct.
+     * @param external_metadata JSON-formatted external metadata.
+     * @param bytes_of_external_metadata The length of @p custom_metadata in
+     * bytes, including the null terminator.
+     * @return ZarrStatus_Success on success, or an error code on failure.
+     */
+    ZarrStatus ZarrStreamSettings_set_custom_metadata(
+      ZarrStreamSettings* settings,
+      const char* external_metadata,
+      size_t bytes_of_external_metadata);
+
+    /**
+     * @brief Get the store path for the Zarr stream.
+     * @param settings The Zarr stream settings struct.
+     * @return The store path for the Zarr stream, or NULL on failure.
+     */
+    const char* ZarrStreamSettings_get_store_path(
+      const ZarrStreamSettings* settings);
+
+    /**
+     * @brief Get the S3 settings for the Zarr stream.
+     * @param settings The Zarr stream settings struct.
+     * @return The S3 settings for the Zarr stream, or an uninitialized struct on
+     * failure.
+     */
+    ZarrS3Settings ZarrStreamSettings_get_s3_settings(
```

Comment:
This illustrates why we should eliminate accessors, even if we opt for the public API to manage the settings object and configure it through accessor functions. If I've already provided S3 store keys, why would I need to retrieve them? Any scenario requiring this retrieval could potentially be considered a security risk.